content/docs/18-knowledgebase/00-how-to/01-how-to-setup-kube-api-audit-logging.md

```diff
@@ -0,0 +1,160 @@
+  - where was it observed?
+  - from where was it initiated?
+  - to where was it going?
+
+## What are some scenarios where you would need to enable kube-api audit logging?
+
+- to meet internal security controls
+- to ensure that the actions of individual users can be uniquely traced back to a specific user
+- to debug an issue where resources are being modified by an unknown application
+
+
+## Performance considerations when enabling audit logging
+
+The audit logging feature increases the memory consumption of the API server because some context required for auditing is stored for each request. Memory consumption depends on the audit logging configuration.
+
+
+## Logging all API requests at the metadata level
+
+To start, we will use the [minimal audit policy file](#creating-the-audit-policy-file) to log all requests at the `metadata` level. 
+
+The output will be written to the log file located at `/var/log/kubernetes/audit/audit.log`, which should be sent to and ingested by a logging appliance (ex. Splunk). 
+
+The `/var/log/kubernetes/audit/` directory should be backed by persistent storage to ensure that during an unexpected outage of the node, any unshipped audit logs will not be lost.  
+
+In this example, the logging appliance should be configured to filter the incoming audit logs. 
+
+## Creating the Audit Policy File
+
+On a control-plane node in the target cluster, run the following command to create your audit policy file. 
+
+**NB.** This example configuration enables log truncation, compression, batching, and retention.
+
+```shell
+cat << EOF > /etc/kubernetes/audit-policy.yaml
+# Log all requests at the Metadata level.
+apiVersion: audit.k8s.io/v1
+kind: Policy
+rules:
+- level: Metadata
+EOF
+```
+
+## Configuring the kube-apiserver manifest
+
+Append the following yaml configuration to your kube-apiserver manifest.
+
+```yaml
+volumeMounts:
+  - mountPath: /etc/kubernetes/audit-policy.yaml
+    name: audit
+    readOnly: true
+  - mountPath: /var/log/kubernetes/audit/
+    name: audit-log
+    readOnly: false
+volumes:
+- name: audit
+  hostPath:
+    path: /etc/kubernetes/audit-policy.yaml
+    type: File
+- name: audit-log
+  hostPath:
+    path: /var/log/kubernetes/audit/
+    type: DirectoryOrCreate
+```
+
+## Configuring kube-apiserver options 
+
+Add the following kube-apiserver flags.
+
+```shell
+    --audit-policy-file=/etc/kubernetes/audit-policy.yaml 
+    --audit-log-path=/var/log/kubernetes/audit/audit.log
+    --audit-log-batch-buffer-size 10000
+    --audit-log-batch-max-size 5
+    --audit-log-compress
+    --audit-log-format "json" 
+    --audit-log-maxage 30
+    --audit-log-maxbackup 100
+    --audit-log-maxsize 50
+    --audit-log-mode "batch"
+    --audit-log-truncate-enabled
+    --audit-log-truncate-max-batch-size 10485760
+    --audit-log-truncate-max-event-size 102400
+    --audit-log-version "audit.k8s.io/v1"
+```
+
+**NB** Descriptions are included below for each option set.
```

Comment:
Instead of the text below, let's add this to a table to improve the readability. We can also avoid the callout in this line.